### DISCUSSION TITLE
The parameters in the default .yalm file cannot be executed, the default default.yalm file is used

### DISCUSSION THREAD

**Discussion by litelpage** [ultralytics/ultralytics#6337] [Q&A]
For example, if I modify the yaml file of the training data, the coco file will still be downloaded during training, and the same problem is the problem for specifying the training data in the train.py file

**Comment by pderrenger** [comment]
@litelpage hello! It seems like you're experiencing an issue where your custom dataset configuration is not being recognized, and the default COCO dataset is being downloaded instead.

To address this, please ensure the following:

1. Your custom dataset YAML file is correctly formatted and saved in the correct directory.
2. You are correctly specifying the path to your custom dataset YAML file when initializing the `YOLO` object or calling the `train()` method.
3. There are no typos or errors in the YAML file that might cause the parser to default back to the COCO dataset.

Here's an example of how to specify your custom dataset YAML file when training:

```python
from ultralytics import YOLO

# Initialize YOLO model and specify the path to your custom dataset YAML
model = YOLO('yolov8n.pt')
results = model.train(data='path/to/your/custom_dataset.yaml', epochs=5)
```

Make sure that `'path/to/your/custom_dataset.yaml'` is the correct path to your dataset configuration file. If you continue to face issues, please check the YAML file for any syntax errors and ensure it follows the expected structure as outlined in the documentation.

If the problem persists, you can provide more details in a GitHub issue, and we'll be happy to help you troubleshoot further. 🛠️
